Bottom: 6e44f7700a0225dd43f6fd2f0352a7c3a2ddba29
Top:    5337cc5e89eb633b65e8a62bc9bfaa5491a04059
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-06 17:23:13 +0000

Workaround crash where contour is passed as NULL to poly_PreContour










---

diff --git a/src/polygon.c b/src/polygon.c
index 43432bc..7410329 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -247,6 +247,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
